crates/consensus/src/block/header.rs

```diff
@@ -214,10 +216,41 @@ impl Header {
     /// Returns `None` if `excess_blob_gas` is None.
     ///
     /// See also [Self::next_block_excess_blob_gas]
+    #[deprecated(note = "Use `next_block_blob_fee_with_fraction` to account for update fractions")]
     pub fn next_block_blob_fee(&self) -> Option<u128> {
         Some(eip4844::calc_blob_gasprice(self.next_block_excess_blob_gas()?))
     }
 
+    /// Returns the blob fee for the next block according to the EIP-4844 spec,
+    /// applying the provided update fraction.
+    ///
+    /// If no fraction is provided, the default is `BLOB_BASE_FEE_UPDATE_FRACTION_ELECTRA`.
+    ///
+    /// # Arguments
+    ///
+    /// - `update_fraction`: Optional fraction (between 0 and 1) that adjusts the calculation.
+    ///
+    /// # Returns
+    ///
+    /// - `Some(u128)` containing the calculated blob fee if `next_block_excess_blob_gas` is `Some`.
+    /// - `None` if `next_block_excess_blob_gas` is `None`.
+    ///
+    /// # Panics
+    ///
+    /// - Panics if `update_fraction` is provided and not within the range [0.0, 1.0].
+    pub fn next_block_blob_fee_with_fraction(&self, update_fraction: Option<f64>) -> Option<u128> {
+        let fraction = update_fraction.unwrap_or(BLOB_BASE_FEE_UPDATE_FRACTION_ELECTRA);
+
+        if !(0.0..=1.0).contains(&fraction) {
+            panic!("update_fraction must be between 0.0 and 1.0");
+        }
+
+        let excess_blob_gas = self.next_block_excess_blob_gas()?;
+        let adjusted_blob_gas = (excess_blob_gas as f64 * fraction) as u64;
+
+        Some(eip4844::calc_blob_gasprice(adjusted_blob_gas))
```

Comment:
we should introduce an additional helper like `calc_blob_gasprice_with_update_fraction` and use it here

Reply:
> we should introduce an additional helper like `calc_blob_gasprice_with_update_fraction` and use it here

thanks for your review!